examples/layer-surface.c

```diff
@@ -0,0 +1,434 @@
+#define _XOPEN_SOURCE
+#define _POSIX_C_SOURCE 200809L
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include <EGL/egl.h>
+#include <GLES2/gl2.h>
+#include <wayland-client-protocol.h>
+#include <wayland-client.h>
+#include <wayland-egl.h>
+#include <wayland-server.h>
+#include <linux/input-event-codes.h>
+#include "surface-layers-client-protocol.h"
+
+struct wl_display *display = NULL;
+struct wl_compositor *compositor = NULL;
+struct wl_output *output = NULL;
+struct surface_layers *layers = NULL;
+
+struct wl_surface *surface;
+struct layer_surface *layer;
+struct wl_egl_window *egl_window;
+struct wl_region *region;
+struct wl_callback *callback;
+
+struct wl_seat *seat;
+struct wl_pointer *pointer;
+struct wl_keyboard *keyboard;
+
+uint32_t width = 300, height = 200;
+
+EGLDisplay egl_display;
+EGLConfig egl_conf;
+EGLSurface egl_surface;
+EGLContext egl_context;
+
+static void draw() {
+	if (eglMakeCurrent(egl_display, egl_surface, egl_surface, egl_context)) {
+		fprintf(stderr, "Made current\n");
+	} else {
+		fprintf(stderr, "Made current failed\n");
+	}
+
+	glClearColor(1.0, 1.0, 0.0, 0.1);
+	glClear(GL_COLOR_BUFFER_BIT);
+	glFlush();
```

Comment:
It's a very minor point, but this call is not needed. eglSwapBuffers will do it for us.